SignalServiceKit/src/Messages/OWSMessageSender.m

```diff
@@ -1130,32 +1131,39 @@ - (void)messageSendDidFail:(TSOutgoingMessage *)message
         }
         case 409: {
             // Mismatched devices
-            DDLogWarn(@"%@ Mismatch Devices for recipient: %@", self.logTag, recipient.uniqueId);
+            DDLogWarn(@"%@ Mismatched devices for recipient: %@", self.logTag, recipient.uniqueId);
 
-            NSError *error;
-            NSDictionary *serializedResponse =
-                [NSJSONSerialization JSONObjectWithData:responseData options:0 error:&error];
-            if (error) {
+            NSError *_Nullable error = nil;
+            NSDictionary *_Nullable responseJson = nil;
+            if (responseData) {
+                responseJson = [NSJSONSerialization JSONObjectWithData:responseData options:0 error:&error];
+            }
+            if (error || !responseJson) {
                 OWSProdError([OWSAnalyticsEvents messageSenderErrorCouldNotParseMismatchedDevicesJson]);
                 [error setIsRetryable:YES];
```

Comment:
I think it's pretty unlikely that retrying would succeed, but I won't change this.